ledger/src/blockstore.rs

```diff
@@ -5315,6 +5315,32 @@ pub mod tests {
                 false
             );
 
+            // Ensure that an empty shred (one with no data) would get inserted. Such shreds
+            // may be used as signals (broadcast does so to indicate a slot was interrupted)
+            // Reuse shred5's header values to avoid a false negative result
+            let empty_shred = Shred::new_from_data(
+                shred5.common_header.slot,
+                shred5.common_header.index,
+                shred5.data_header.parent_offset,
+                None, // data
+                true, // is_last_data
+                true, // is_last_in_slot
+                0,    // reference_tick
+                shred5.common_header.version,
+                shred5.common_header.fec_set_index,
+            );
+            assert_eq!(
+                blockstore.should_insert_data_shred(
+                    &empty_shred,
+                    &slot_meta,
+                    &HashMap::new(),
+                    &last_root,
+                    None,
+                    false
+                ),
+                true
+            );
```

Comment:
I think after this assert, we also need to set `shred.data_header.size` to 0, and confirm that it then returns false.

Currently, I think this is short circuiting at this line:
https://github.com/solana-labs/solana/blob/63c56b57a/ledger/src/blockstore.rs#L1287-L1289

and does not even hit `if shred.data_header.size == 0 {` check.

Reply:
So I thought the same thing the first time I looked at that snippet; however, that statement is only assigning a value to `last_in_slot` (with the added debug statement), not returning from the entire function.
```
        let last_in_slot = if shred.last_in_slot() {
            debug!("got last in slot");
            true
        } else {
            false
        };
```

Reply:
oh, you are right.
so maybe just set the `shred.data_header.size` to 0 after this assert, and confirm that it then returns false.